vllm/model_executor/models/molmo.py

```diff
@@ -0,0 +1,1130 @@
+
+        embedding_weight = dict()
+        projector_weight = dict()
+        for name, loaded_weight in weights:
+            log.info(f"Original name: {name}")
+            if "rotary_emb.inv_freq" in name:
+                log.info(f"Skipping {name}")
+                continue
+            if self.config.tie_word_embeddings and "lm_head.weight" in name:
+                continue
+
+            if "wte.embedding" in name:
+                embedding_weight["embedding"] = loaded_weight
+                continue
+                
+            if "wte.new_embedding" in name:
+                embedding_weight["new_embedding"] = loaded_weight
+                continue
+           
+            if "vision_backbone" in name:
+                if name.startswith("model"):
+                    name = name[len("model."):]
+                if 'image_projector' in name:
+                    if 'w1' in name:
+                        projector_weight['gate_proj'] = loaded_weight
+                    elif 'w3' in name:
+                        projector_weight['up_proj'] = loaded_weight
+                    elif 'w2' in name:
+                        projector_weight['down_proj'] = loaded_weight
+                    else:
+                        raise ValueError(f"Unexpected projector weight: {name}")
+                    continue
+            else:
+                if "ln_f.weight" in name:
+                    name = "model.norm.weight"
+
+                if "transformer.blocks" in name:
+                    name = name.replace("transformer.blocks", "layers")
+    
+                if "attn_out" in name:
+                    name = name.replace("attn_out", "self_attn.o_proj")
+                
+                if "att_proj" in name:
+                    name = name.replace("att_proj", "self_attn.qkv_proj")
+                
+                if 'q_norm' in name:
+                    name = name.replace("q_norm", "self_attn.q_norm")
+                
+                if 'k_norm' in name:
+                    name = name.replace("k_norm", "self_attn.k_norm")
+
+                if "ff_proj" in name:
+                    name = name.replace("ff_proj", "mlp.gate_up_proj")
+                    assert 'weight' in name
+                    up_weight, gate_weight = loaded_weight.chunk(2, dim=0)
+                    loaded_weight = torch.cat([gate_weight, up_weight], dim=0)
+
+                if "ff_out" in name:
+                    if "layers" in name:
+                        name = name.replace("ff_out", "mlp.down_proj")
+                    else:
+                        # lm head
+                        name = name.replace("model.transformer.ff_out", "lm_head")
+                    
+                if "attn_norm" in name:
+                    name = name.replace("attn_norm", "input_layernorm")
+                
+                if "ff_norm" in name:
+                    name = name.replace("ff_norm", "post_attention_layernorm")
```

Comment:
Since these are simple renaming, you can add them to the mapping I suggest above too.